pkg/client/typed_client.go

```diff
@@ -22,6 +22,10 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+var _ Reader = &typedClient{}
```

Comment:
Wouldn't it be easier to verify that `&typedClient{}` is a `Client` rather than that it is everything a `Client` is composed of?

Reply:
tried that, although it doens't actually satisfy that interface, because it's missing Status() to get a status client, and RESTMapper() and Scheme()

I could backfill those if we want, but not sure if it's worth it

Reply:
Ah okay, no its fine like that then